web/src/alerts.ts

```diff
@@ -21,47 +22,57 @@ function hasAlert(resource: Resource) {
 
 //These functions determine what kind of error has occurred based on information about
 //the resource - return booleans
-function crashRebuild(resource: Resource): boolean {
-  return (
-    resource.BuildHistory.length > 0 && resource.BuildHistory[0].IsCrashRebuild
-  )
+
+//Errors for K8s Resources
+function crashRebuild(r: Resource): boolean {
+  return r.BuildHistory.length > 0 && r.BuildHistory[0].IsCrashRebuild
 }
 
-function podStatusHasError(resource: Resource) {
-  let podStatus = resource.ResourceInfo.PodStatus
-  let podStatusMessage = resource.ResourceInfo.PodStatusMessage
+function podStatusHasError(r: Resource) {
+  let rInfo = <K8sResourceInfo>r.ResourceInfo
+  let podStatus = rInfo.PodStatus
+  let podStatusMessage = rInfo.PodStatusMessage
   if (podStatus == null) {
     return false
   }
   return podStatusIsError(podStatus) || podStatusMessage
 }
 
-function podRestarted(resource: Resource) {
-  return resource.ResourceInfo.PodRestarts > 0
+function podRestarted(r: Resource) {
+  let rInfo = <K8sResourceInfo>r.ResourceInfo
+  return rInfo.PodRestarts > 0
 }
 
+// Errors for both DC and K8s Resources
 function buildFailed(resource: Resource) {
   return (
     resource.BuildHistory.length > 0 && resource.BuildHistory[0].Error !== null
   )
 }
 
-function numberOfAlerts(resource: Resource): number {
-  return getResourceAlerts(resource).length
+function isK8sResourceInfo(
+  resourceInfo: K8sResourceInfo | DCResourceInfo
+): resourceInfo is K8sResourceInfo {
```

Comment:
Woah, this function is crazy. What is happening here? I want to learn. :)

Reply:
prettier made it ugly >:( but basically it takes in a resourceInfo of either type and checks if a field is defined in there, so in here it checks in the resourceInfo if podName is defined (which it would be for a K8s resource and not a DC resource). Then it returns a boolean based on that :)

Reply:
What is `is` for?

Reply:
its called a "type predicate" found it on here: https://www.typescriptlang.org/docs/handbook/advanced-types.html